learning_phase/go-chat/client/client.go

```diff
@@ -122,15 +222,45 @@ func (cli *client) Run(ctx context.Context, main_term_chan chan bool) {
 	2. main_term_chan - write to this channel when the function has completely exited to inform main
 	*/
 
-	/** 
+	/**
 	Called by main.go
 	Attempt to create a TCP connection to the server, authenticate client and check for errors returned by server.
 	If there are no errors, create a Scanner to read user inputs (you can use NewScanner in bufio package).
 	Then you will need to listen for user input as well as messages delivered by the server, using the above defined functions.
-	Note: there are two ways through which the program termination will occur - when the user exits the program 
+	Note: there are two ways through which the program termination will occur - when the user exits the program
 	(the signal will be caught by main() and propogated to this function) as well as if the server tells the client to terminate
 	for whatever reason. You should listen for and handle both.
 	*/
+	var addr string
+	fmt.Println("\n Hi " + cli.Username + "! Enter host to connect to: ")
+	fmt.Scanln(addr)
+	cli.ServerHost = "0.0.0.0" + addr
```

Comment:
No need to hardcode 0.0.0.0, cause it will only look for listening sockets bound to the given port at all IP addresses of the local machine. It won't work if the server is somewhere else in the network